components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -101,7 +105,9 @@ public StorageTxImpl(final BlobTx blobTx,
     this.assetEntityAdapter = checkNotNull(assetEntityAdapter);
     this.hook = checkNotNull(hook);
 
-    db.begin(TXTYPE.OPTIMISTIC);
+    if (!userManagedDb) {
+      db.begin(TXTYPE.OPTIMISTIC);
```

Comment:
As discussed on the phone, the transaction-control aspect of StorageTx needs to be made consistent.  The close() and rollback() methods are still dealing with transactions, so the constructor should be calling db.begin() regardless of the user-managed mode.